lib/plugins/aws/deploy/lib/uploadArtifacts.js

```diff
@@ -6,24 +6,33 @@ const filesize = require('filesize');
 const path = require('path');
 
 module.exports = {
-  uploadCloudFormationFile() {
-    this.serverless.cli.log('Uploading CloudFormation file to S3...');
+  uploadCloudFormationFiles() {
+    this.serverless.cli.log('Uploading CloudFormation files to S3...');
 
-    const compiledTemplateFileName = 'compiled-cloudformation-template.json';
+    const serverlessDirPath = path.join(this.serverless.config.servicePath, '.serverless');
+    const cfFileNameRegex = this.provider.naming.getCloudFormationFileNameRegex();
+    const filesInServiceDir = fs.readdirSync(serverlessDirPath);
+    const cfFiles = filesInServiceDir.filter((file) => file.match(cfFileNameRegex));
 
-    const body = JSON.stringify(this.serverless.service.provider.compiledCloudFormationTemplate);
-    const params = {
-      Bucket: this.bucketName,
-      Key: `${this.serverless.service.package.artifactDirectoryName}/${compiledTemplateFileName}`,
-      Body: body,
-      ContentType: 'application/json',
-    };
+    const uploadPromises = [];
+    cfFiles.forEach((file) => {
+      const body = fs.createReadStream(path.join(serverlessDirPath, file));
```

Comment:
You must make sure that the in-memory template is used here - or dump the in-memory representation just before the upload! The problem is, that the file is persisted after the package step, but plugins that modify it during the deployment phase typically only modify the in-memory representation.
Loading and deploying the stored file here without having it persisted here in the same file, would remove all plugin changes to the template.
Examples here are the alias plugin and the warmup plugin.